File: java/hello/src/main/Circle.java
package main;

public class Circle{
	private int x = 0;
	private int y = 0;
	private float r = 1;

	public Circle(float r){ 
		this.r = r;
	}   

	public void setRadius(float r){ 
		if (r > 0)  this.r = r;
		else        this.r = 1;
	}   

	public float getRadius(){
		return this.r;
	}   

	public double getArea(){
		return this.r * this.r * java.lang.Math.PI;                                                                  
	}   
}

File: c/07_object/circle.c
#include <stdio.h>
#include "circle.h"

// member method 
float getArea(struct Circle *this){
    return 3.14 * this->radius * this->radius;
}

void setRadius(struct Circle *this, float arg){
    this->radius = arg;
}

float getRadius(struct Circle *this){
    return this->radius;
}

// constructure
void newCircle(struct Circle *this, float r){
    this->new = newCircle;
    this->area = getArea;
    this->setRadius = setRadius;
    this->getRadius = getRadius;
    //this->radius = r;
    setRadius(this, r);
}




